Update API docs for `required_fields` (#5190)
elastic/security-docs: docs/detections/api/rules/rules-api-create.asciidoc

|references |String[] |Array containing notes about or references to
relevant information about the rule. Defaults to an empty array.

> |required_fields |Object[] a| Elasticsearch fields and their types that need to be present for the rule to function. The object has these fields:
> 
>  * `name` (string, required): Field name
>  * `type` (string, required): Field type
>  * `ecs` (boolean, optional): Indicates whether the field is {ecs-ref}[ECS-compliant]. This property is only present in responses. Its value is computed based on field's name and type.
> 
> *NOTE*: The value of `required_fields` does not affect the rule's behavior. It can be thought of as a way to document the fields that the rule expects to be present in the data. Specifying `required_fields` incorrectly won't cause the rule to fail.

Reviewer: ```suggestion
|required_fields |Object[] a| {es} fields and their types that need to be present for the rule to function. The object has these fields:

 * `name` (string, required): The field's name.
 * `type` (string, required): The field's data type.
 * `ecs` (Boolean, optional): Indicates whether the field is {ecs-ref}[ECS-compliant]. This property is only present in responses. Its value is computed based on field's name and type.

*NOTE*: The value of `required_fields` does not affect the rule's behavior, and specifying it incorrectly won't cause the rule to fail. Use `required_fields` as an informational property to document the fields that the rule expects to be present in the data. 
```
